apps/frontend/src/components/review/TechnicalBulkReassignModal.tsx

```diff
@@ -0,0 +1,309 @@
+
+  const reviewsSortedByInstrument: GroupedReviews[] = reviews.reduce(
+    (acc: GroupedReviews[], review) => {
+      const instrumentId = review.review.instrumentId;
+      const instrumentName = review.instrument?.name;
+
+      proposalCount.set(
+        review.proposal.proposalId,
+        proposalCount.get(review.proposal.proposalId)
+          ? (proposalCount.get(review.proposal.proposalId) as number) + 1
+          : 1
+      );
+
+      const instrumentGroup = acc.find(
+        (group) => group.instrumentId === instrumentId
+      );
+
+      if (instrumentGroup) {
+        instrumentGroup.reviews.push(review);
+      } else {
+        acc.push({
+          instrumentId,
+          instrumentName,
+          reviews: [review],
+        });
+      }
+
+      return acc;
+    },
+    []
+  );
+
+  const proposalsWithMultipleInstrument: string[] = [];
+  proposalCount.forEach((count, prop) => {
+    if (count > 1) {
+      proposalsWithMultipleInstrument.push(prop);
+    }
+  });
+
+  const updateReviews = () => {
+    // Map over current instruments as there is a edge case where the map includes instruments that are no longer present
+    instruments?.map((inst) => {
+      const proposals = reviewsSortedByInstrument
+        .find((group) => group.instrumentId === inst.id)
+        ?.reviews.map((rev) => rev.proposal.proposalPk);
+
+      newReviwersMap.get(inst.id) &&
+        proposals &&
+        api().updateTechnicalReviewAssignee({
+          instrumentId: inst.id,
+          proposalPks: proposals,
+          userId: newReviwersMap.get(inst.id) as number,
+        });
+    });
+    setReviews(true, true);
+  };
+
+  const instrumentColumns: Column<GroupedReviews>[] = [
+    {
+      title: t('instrument'),
+      field: 'instrumentName',
+      sorting: false,
+      render: (data) => data.instrumentName,
+    },
+    {
+      title: 'New Technial Reviewer',
+      sorting: false,
+      render: (data) => {
+        const instrument = instruments?.find(
+          (inst) => inst.id === data.instrumentId
+        );
+
+        const userList = instrument ? [...instrument?.scientists] : [];
+        if (
+          !userList.find(
+            (user) => user.id === instrument?.instrumentContact?.id
+          )
+        ) {
+          instrument?.instrumentContact &&
+            instrument &&
+            userList.push(instrument?.instrumentContact);
+        }
+
+        return (
+          <Autocomplete
+            id={`user-list-${data.instrumentId}`}
+            options={userList}
+            renderInput={(params) => (
+              <TextField
+                {...params}
+                label="Technical reviewer"
+                margin="none"
+                data-cy="tech-reviewer-" //need?
+              />
+            )}
+            getOptionLabel={(option) => getFullUserName(option)}
+            onChange={(_event, newValue) => {
+              if (instrument && newValue) {
+                setNewReviwersMap(
```

Comment:
State objects are meant to be immutable, so must set the state to the new map object. If you know a better way let me know